From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shelley Vohr <shelley.vohr@gmail.com>
Date: Wed, 16 Oct 2019 09:07:57 -0700
Subject: fix: windows compilation on libuv setsockopt

Upstreamed in https://github.com/libuv/libuv/pull/2520.
This patch should be removed when Node.js rolls onto libuv v1.33.0.

diff --git a/deps/uv/src/win/tcp.c b/deps/uv/src/win/tcp.c
index 81e48136a3b9ef13b1b95d87a68ab3ba98f9aeb9..41f8fcbe8361dc7c7a145b38094745cb0fcfa350 100644
--- a/deps/uv/src/win/tcp.c
+++ b/deps/uv/src/win/tcp.c
@@ -556,7 +556,7 @@ int uv_tcp_close_reset(uv_tcp_t* handle, uv_close_cb close_cb) {
   if (handle->flags & UV_HANDLE_SHUTTING)
     return UV_EINVAL;
 
-  if (0 != setsockopt(handle->socket, SOL_SOCKET, SO_LINGER, &l, sizeof(l)))
+  if (0 != setsockopt(handle->socket, SOL_SOCKET, SO_LINGER, (const char*)&l, sizeof(l)))
     return uv_translate_sys_error(WSAGetLastError());
 
   uv_close((uv_handle_t*) handle, close_cb);
